beacon_node/beacon_chain/src/custody_context.rs

```diff
@@ -164,12 +227,18 @@ impl<E: EthSpec> CustodyContext<E> {
 
     pub fn new_from_persisted_custody_context(
         ssz_context: CustodyContextSsz,
-        is_supernode: bool,
-    ) -> Self {
-        CustodyContext {
+        node_custody_type: NodeCustodyType,
+    ) -> Result<Self, String> {
+        if ssz_context.persisted_is_supernode && node_custody_type != NodeCustodyType::Supernode {
```

Comment:
So we might want a DB schema change to store the `NodeCustodyType`? Then we could enforce that the node can't change "down", i.e. supernode to semi-supernode/full, or semi-supernode to full?

Does anything break if we don't block the transition from semi-supernode to full? Because this check already covers both supernode cases.